docs/adguard-for-android/solving-problems/background-work.md

```diff
@@ -0,0 +1,395 @@
+
+1. Switch Power Saving Modes to Off
+2. Choose the following options: *Saving Power in The Background > Choose apps > Select your app > Background Settings > No restrictions*
+
+#### App battery saver
+
+*Security > Battery > App Battery Saver > AdGuard > No restriction*
+
+#### App pinning
+
+To set up AdGuard's background work for Xiaomi devices you should pay attention to Battery and Permissions.
+
+- Tap on the *Recent tasks* button and swipe AdGuard down to make options *visible* (as shown on the screenshot):
+
+![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomirecent.png)
+
+- Tap on the *lock* icon. This will stop Xiaomi from closing AdGuard automatically. It should look like this:
+
+![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomilocked.png)
+
+- Go to *"Battery"*
+
+- Select *"battery saver"* app 
+
+- Find and select AdGuard
+
+- Set up the following *"Background settings"*:
+
+![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomirest.png)
+
+- Go to *"Permissions"*
+
+- Select *"Autostart"*
+
+- Make sure that autostart function is enabled for AdGuard:
+
+![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomiautostart.png)
+
+## Samsung
+
+#### Galaxy S9 / S10
+
+Battery optimization is **turned on by default**. At some unclear moments (maybe on app update, OS update), the settings also do revert back to the defaults, forcing you to turn them off again and again.
+
+#### Sleeping apps
+
+Sleeping apps menu is the sniper’s nest for Samsung’s app killing policies. Make sure to follow the instructions very carefully to prevent the apps from being killed.
+
+Checklist:
+
+- Disable **Put unused apps to sleep**
+- Disable **Auto-disable unused apps**
+- Remove your app from the list of sleeping apps
+- Disable background restrictions for your app
+
+1. Start **Device care** from phone settings
+
+![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung1en.png)
+
+2. Tap **Battery**
+
+![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung2en.png)
+
+
+3. Tap the **3-dot menu** > **Settings**
+
+![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung3en.png)
+
+4. Disable all toggles (except Notifications)
+5. Tap **Sleeping apps**
+
+![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung45en.png)
+
+6. **Wake up** all the apps using the trashcan icon
+
+![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung6en.png)
+
+> **Note**: Make sure that Put unused apps to sleep and Auto-disable unused apps is disabled. Otherwise Samsung will put your apps back to sleep after a few days (3 by default) even if you have woken them up manually!
```

Comment:
```suggestion
> **Note**: Make sure that `Put unused apps to sleep` and `Auto-disable unused apps` are disabled. Otherwise, Samsung will put your apps back to sleep after a few days even if you have woken them up manually.
```